source/layoutex/ParagraphLayout.cpp

```diff
@@ -995,7 +995,19 @@ le_int32 ParagraphLayout::previousBreak(le_int32 charIndex)
         UCharCharacterIterator *iter = new UCharCharacterIterator(fChars, fCharCount);
         UErrorCode status = U_ZERO_ERROR;
 
-        fBreakIterator = BreakIterator::createLineInstance(thai, status);
+        // Set the break iterator; default is line breaks
+        if (kind == UBreakIteratorType::UBRK_CHARACTER) {
+            fBreakIterator = BreakIterator::createCharacterInstance(thai, status);
+        } else if (kind == UBreakIteratorType::UBRK_WORD) {
+            fBreakIterator = BreakIterator::createWordInstance(thai, status);
+        } else if (kind == UBreakIteratorType::UBRK_SENTENCE) {
+            fBreakIterator = BreakIterator::createSentenceInstance(thai, status);
+        } else if (kind == UBreakIteratorType::UBRK_TITLE) {
+            fBreakIterator = BreakIterator::createTitleInstance(thai, status);
+        } else {
+            fBreakIterator = BreakIterator::createLineInstance(thai, status);
```

Comment:
any idea what's up with that locale?

Reply:
No clue. Tried looking into it but didn't find any info 😬

Reply:
weird.